packages/toolkit/src/login/webview/vue/backend.ts

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import globals from '../../../shared/extensionGlobals'
+import { VueWebview } from '../../../webviews/main'
+import { Region } from '../../../shared/regions/endpoints'
+import { ToolkitError } from '../../../shared/errors'
+import { CancellationError } from '../../../shared/utilities/timeoutUtils'
+import { trustedDomainCancellation } from '../../../auth/sso/model'
+import { handleWebviewError } from '../../../webviews/server'
+import { InvalidGrantException } from '@aws-sdk/client-sso-oidc'
+import { SsoConnection } from '../../../auth/connection'
+import { Auth } from '../../../auth/auth'
+import { StaticProfile, StaticProfileKeyErrorMessage } from '../../../auth/credentials/types'
+
+export type AuthError = { id: string; text: string }
+export const userCancelled = 'userCancelled'
+
+export abstract class CommonAuthWebview extends VueWebview {
+    public override id: string = 'aws.AmazonCommonAuth'
+    public override source: string = 'src/login/webview/vue/index.js'
+
+    public getRegions(): Region[] {
+        return globals.regionProvider.getRegions().reverse()
+    }
+
+    /**
+     * This wraps the execution of the given setupFunc() and handles common errors from the SSO setup process.
+     *
+     * @param methodName A value that will help identify which high level function called this method.
+     * @param setupFunc The function which will be executed in a try/catch so that we can handle common errors.
+     * @returns
+     */
+    async ssoSetup(methodName: string, setupFunc: () => Promise<any>): Promise<AuthError | undefined> {
```

Comment:
Can we make this static and have `show.ts` for the current auth page use this instead? It will help to have this in a single spot

Reply:
Can you elaborate? There is no `show.ts` in the new login code path. 

I can import this function from current auth page to avoid duplicating code.

Reply:
I can further improve this in next PR.